packages/dds/tree/src/events/events.ts

```diff
@@ -59,7 +59,10 @@ export type TransformEvents<E extends Events<E>, Target extends IEvent = IEvent>
     : never;
 
 /**
- * An object which allows the registration of listeners so that subscribers can be notified when an event happens
+ * An object which allows the registration of listeners so that subscribers can be notified when an event happens.
+ * Clients can subscribe to events via the `on` method and unsubscribe by invoking the return value of `on`.
+ * Classes wishing to emit events via this interface may implement this interface by creating a private {@link EventEmitter}
+ * or by extending {@link EventEmitter} directly.
```

Comment:
The composition approach works even if not using classes, so it might be better to rephrase this.

Also I think a blank line between the definition of this type and the optional implementation guidance would be good (so they are separate paragraphs).

```suggestion
 *
 * {@link EventEmitter} can help implement this interface via delegation or extension.
```